refinery/core/models.py

```diff
@@ -214,6 +214,11 @@ def duplicate_slug_check(self):
     def clean(self):
         # Check if model being saved/altered in Django Admin has a slug
         # duplicated elsewhere.
+
+        # Removes trailing/leading white spaces
```

Comment:
i would start the method with

    if not self.slug:
        return

then avoid the `if self.slug` later